flax/nn/base.py

```diff
@@ -931,29 +940,62 @@ def store(self, value):
     Returns:
       The previous value stored in the collection or None.
     """
-    _top_frame('store')
+    frame = _top_frame('store')
     if not self._mutable:
       raise ValueError('Collection is not mutable. Use the `mutate` method to'
                        'create a mutable copy.')
-    assert len(_module_stack) > self._anchor
-
     # the root of a Collection is the first module scope that gets created
     # inside the mutate scope of the Collection. By allowing only one unique
     # root scope we guarantee that state is not accidentally shared
     # between different models. When a user specifies an explicit name we can
     # distinguish models and a collection can have multiple roots.
-    root = _module_stack[self._anchor]
+    if frame is self._anchor:
```

Comment:
is the use of `is` instead of `==` here important? if so that's a sharp edge that a lot of people won't know about.